agent/sidecar_service.go

```diff
@@ -10,12 +10,31 @@ import (
 	"github.com/hashicorp/consul/agent/structs"
 )
 
-func sidecarServiceID(serviceID string) string {
-	return serviceID + "-sidecar-proxy"
+const sidecarIDSuffix string = "-sidecar-proxy"
+
+func sidecarIDFromServiceID(serviceID string) string {
+	return serviceID + sidecarIDSuffix
 }
 
+// reverses the sidecarIDFromServiceID operation
 func serviceIDFromSidecarID(sidecarServiceID string) string {
-	return strings.Split(sidecarServiceID, "-")[0]
+	// we only want to un-append "-sidecar-proxy" from the very end
```

Comment:
I think we can simply call `strings.TrimSuffix` here, right? https://pkg.go.dev/strings#TrimSuffix

Reply:
great find! thank you! that makes it much cleaner